File: internships/yandex/shad/1/python/2.py
arr = []
n = int(input())

for _ in range(n):
    arr.append(list(map(int, input().split())))


res = 0

for i in range(n):
    for j in range(n):
        if i != j:
            for a in range(6):
                for b in range(6):
                    res += (arr[i][a] + arr[j][b])**3

print(res/(36*(n - 1)*n))

File: internships/yandex/mobile/mobile_summer_2023/tasks/2.swift
var n = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map { Int($0)! }

var res = ""
var temp: Set<Int> = []

for i in 0..<n {
    temp.insert(arr[i])
    
    if temp.contains(n) {
        var add = ""
        
        while let index = temp.firstIndex(of: n) {
            add += "\(temp[index]) "
            temp.remove(n)
            n -= 1
        }
        res += "\(add)\n"
    } else {
        res += "\n"
    }
}

if !temp.isEmpty {
    res += "\(temp.map { String($0) }.joined(separator: " "))\n"
}

print(res)